### PR TITLE
[DNM] Closes #499 - frontend for adding star comments to user profile

### LINKED ISSUE #499: [Feature] See star comments on user profile
Currently the comments that the users submit when giving a star aren't available through the app UI.

These comments should be publicly displayed on the profile of the user that received the star.

### PR DESCRIPTION
**Pull request by kondanna**
Because we are moving away from @types folder and towards graphql/index folder, doing just one import of Star from graphql/index raised Typescript issues on profileSubmissions and its story. 

also changed the import on starsController, which required changing some typeDefs/Star fields to mandatory.
mentos_tserror
TSerror2

### CODE REVIEW

**Comment by kondanna** [comment]
typescript errors

**Comment by Ulisseus** [comment]
Is `ProfileStarComments` supposed to look like [this](https://deploy-preview-582--youthful-thompson-448332.netlify.app/?path=/story/components-profilestarcomments--profile-star-comments)? 
I guess css styles don't work in storybook. You can change webpack config like [this](https://github.com/garageScript/c0d3-app/pull/574/files/a5de199ee54781d28958c6d11fa6b7576bd891ad#diff-c4929098279e148b29c0db30288d3c28fdc4f70d3cf0529641fb59c46a390b32) to fix it.

**Comment by kondanna** [comment]
> Is `ProfileStarComments` supposed to look like [this](https://deploy-preview-582--youthful-thompson-448332.netlify.app/?path=/story/components-profilestarcomments--profile-star-comments)?
> I guess css styles don't work in storybook. You can change webpack config like [this](https://github.com/garageScript/c0d3-app/pull/574/files/a5de199ee54781d28958c6d11fa6b7576bd891ad#diff-c4929098279e148b29c0db30288d3c28fdc4f70d3cf0529641fb59c46a390b32) to fix it.

haha nope, you did update the scss module thing for storybook so it should start looking better soon.

**Comment by Ulisseus** [comment]
This is how current `ProfileStarComments` [looks](https://deploy-preview-582--youthful-thompson-448332.netlify.app/?path=/story/components-profilestarcomments--profile-star-comments). Nice. It is possible to add another story to show full profile page with start comments? It should be under left panel `Charles Ko @cko` in [profile](https://www.c0d3.com/profile/cko), right? What if someone has a lot of stars like 20 or some? Will it overflow or scroll?

As for type errors. There is one automatically generated Star type in `graphql/index` and another one under `@types/lesson`. One has `id` field, another doesn't. I guess the same problem with `lessonChallenges` one type was manually typed another one was autogenerated. Is it possible to switch to codegen generated types only?

**Comment by kondanna** [comment]
> This is how current `ProfileStarComments` [looks](https://deploy-preview-582--youthful-thompson-448332.netlify.app/?path=/story/components-profilestarcomments--profile-star-comments). Nice. It is possible to add another story to show full profile page with star comments? It should be under left panel `Charles Ko @cko` in [profile](https://www.c0d3.com/profile/cko), right? What if someone has a lot of stars like 20 or some? Will it overflow or scroll?
> 
> As for type errors. There is one automatically generated Star type in `graphql/index` and another one under `@types/lesson`. One has `id` field, another doesn't. I guess the same problem with `lessonChallenges` one type was manually typed another one was autogenerated. Is it possible to switch to codegen generated types only?

I'll work on that story with like 20 star comments and see how it looks. right now it should overflow, which isn't ideal. I thought about a scroll bar, but the other components don't have one so maybe it would look out of place. i'll try both to see.

File: __tests__/pages/profile/username.test.js [hunk 1]
```diff
@@ -8,14 +8,10 @@ import { MockedProvider } from '@apollo/client/testing'
 import dummyLessonData from '../../../__dummy__/lessonData'
 import dummySessionData from '../../../__dummy__/sessionData'
 import { useRouter } from 'next/router'
-import expectLoading from '../../utils/expectLoading'
 
 describe('user profile test', () => {
   const { query } = useRouter()
   query['username'] = 'fake user'
-  test('Should render loading spinner if data is not ready', () => {
-    expectLoading(<UserProfile />)
```
Comment (Ulisseus) [inline __tests__/pages/profile/username.test.js:17]: Why this test was removed? It shouldn't cause any problems.

Comment (kondanna) [reply __tests__/pages/profile/username.test.js:17]: I'm putting it back. Thought we were getting rid of those, but I must have misunderstood.

**Comment by kondanna** [comment]
since this has backend code, should wait for prisma integration before merging (see other PR)

File: __tests__/pages/profile/username.test.js (hunk 1 above)
Comment (Ulisseus) [reply __tests__/pages/profile/username.test.js]: We removed snapshot testing (rendering full page and comparing element by element). `expectLoading` simply checks for `loading` message in the document, so it should be fine.

**Comment by Ulisseus** [comment]
Your stories for `userprofile` [don't work](https://deploy-preview-582--youthful-thompson-448332.netlify.app/?path=/story/pages-userprofile--user-profile-zero-comments), it shows server error for me (you can always check it locally with `STORYPORT=xxxx yarn storybook` before pushing it).

Luckily it's possible to deploy vercel preview and check new profiles.  [ggwadera](https://c0d3-app-omfqz87sf-c0d3.vercel.app/profile/ggwadera) with a lot of stars, [you](https://c0d3-app-omfqz87sf-c0d3.vercel.app/profile/cko) with two stars or [me](https://c0d3-app-omfqz87sf-c0d3.vercel.app/profile/uliss) without a single star. Looks great. But lessons css is broken again, so you should fix it. 

You can remove new broken stories (or fix them if you want to).

**Comment by kondanna** [comment]
yeah the profile challenges css is still the old code, so i'll have to pull the latest from upstream.

**Comment by ggwadera** [comment]
That star SVG in the middle of the card seems to be wasting a lot of space, doesn't look very nice to me.

What if you only display the user if there's no comment in the star? Something like this:

image

What do you think?

**Comment by kondanna** [comment]
Looks good, should the heading be "Stars" or "Stars Received"?

**Pull request closed without merging**